packages/ra-core/src/dataProvider/useQueryWithStore.ts

```diff
@@ -201,6 +204,7 @@ export const useQueryWithStore = <State extends ReduxState = ReduxState>(
             queriesThisTick[requestSignature] = new Promise<PartialQueryState>(
                 resolve => {
                     dataProvider[type](resource, payload, options)
+                        // @ts-ignore
```

Comment:
There's one thing I wasn't able to address: ensuring that a dataProvider method always returns a Promise. Hence the ignore